zebra-rpc/src/methods/tests/snapshot/get_block_template_rpcs.rs

```diff
@@ -262,7 +274,62 @@ pub async fn test_responses<State, ReadState>(
         .zcash_deserialize_into()
         .expect("coinbase bytes are valid");
 
-    snapshot_rpc_getblocktemplate("long_poll", get_block_template, coinbase_tx, &settings);
+    snapshot_rpc_getblocktemplate(
+        "long_poll",
+        (*get_block_template).into(),
+        Some(coinbase_tx),
+        &settings,
+    );
+
+    // `getblocktemplate` proposal mode variant
+
+    let get_block_template = tokio::spawn(get_block_template_rpc.get_block_template(Some(
+        get_block_template::JsonParameters {
+            mode: GetBlockTemplateRequestMode::Proposal,
+            data: Some(HexData("".into())),
+            ..Default::default()
+        },
+    )));
+
+    let get_block_template = get_block_template
+        .await
+        .expect("unexpected panic in getblocktemplate RPC task")
+        .expect("unexpected error in getblocktemplate RPC call");
+
+    snapshot_rpc_getblocktemplate("invalid-proposal", get_block_template, None, &settings);
```

Comment:
Nitpick: consistent filenames

```suggestion
    snapshot_rpc_getblocktemplate("invalid_proposal", get_block_template, None, &settings);
```